### PR TITLE
27 implement Board_3x3 class  and HumanPlayer

### LINKED ISSUE #27: Task: implement Board Class + Testing

### PR DESCRIPTION
**Pull request by RefRexi**

### CODE REVIEW

File: src/main/java/game/Board_3x3.java [hunk 1]
```diff
@@ -1,24 +1,101 @@
...
     @Override
     public BoardState getBoardState() {
-        return null;
+        return boardState;
     }
 
     @Override
     public PlayerState getPlayerState(Player player) {
-        return null;
+        return playerState;
```
Comment (srebl) [inline src/main/java/game/Board_3x3.java]: only retuns the state of the last player

File: src/main/java/game/AiPlayer.java
Comment (srebl) [inline src/main/java/game/AiPlayer.java:1]: remove this file

File: src/main/java/game/HumanPlayer.java [hunk 2]
```diff
@@ -3,22 +3,44 @@
 import common.Result;
 
 class HumanPlayer implements Player {
-    HumanPlayer(GameIO io, String name, char mark){
+    private GameIO io;
+    private String name;
+    private char mark;
 
+    // Constructor
+    HumanPlayer(GameIO io, String name, char mark) {
```
Comment (srebl) [inline src/main/java/game/HumanPlayer.java]: add a guard against null values. something like 
HumanPlayer(@NotNull  io,  @NotBlank name, char mark)

File: src/main/java/game/HumanPlayer.java [hunk 3]
```diff
@@ -3,22 +3,44 @@
...
     @Override
     public String getName() {
-        return null;
+        return name;
     }
 
     @Override
-    public Result<Boolean, String> makeMove(Board board) {
-        return null;
+    public boolean makeMove(Board board) {
```
Comment (srebl) [inline src/main/java/game/HumanPlayer.java]: guard against null value

File: src/main/java/game/Board_3x3.java [hunk 4]
```diff
@@ -1,24 +1,101 @@
...
+
+        board[x][y] = player.getMark();
+        updateBoardAndPlayerState(x, y, player.getMark());
+        return true;
     }
 
     @Override
     public char[][] getBoardValues() {
-        return new char[0][];
+        return board;
```
Comment (srebl) [inline src/main/java/game/Board_3x3.java]: Copy the array and return a new one. A Caller might edit the internal array.

File: src/test/java/game/board3X3Test.java
Comment (srebl) [inline src/test/java/game/board3X3Test.java:1]: test method names not conforming to guidelines. not all methods are tested.

File: src/test/java/game/board3X3Test.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Test
+    public void testAddMove() {
+        assertTrue(board.addMove(player1, 0, 0));
+        assertEquals('X', board.getBoardValues()[0][0]);
+    }
+
+    @Test
+    public void testInvalidMove() {
+        assertFalse(board.addMove(player1, 3, 3)); // outside bounds
```
Comment (srebl) [inline src/test/java/game/board3X3Test.java]: use parameterized test

File: src/test/java/game/board3X3Test.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        board.addMove(player1, 0, 0);
+        assertFalse(board.addMove(player1, 0, 0)); // occupied cell
+    }
+
+    @Test
+    public void testWinCondition() {
```
Comment (srebl) [inline src/test/java/game/board3X3Test.java]: test for all win conditions. you could set the private array with a reflection.

**Review by srebl — CHANGES_REQUESTED** [review]
please resolve problems highlighted in the comments. Add tests for HumanPlayer. No JavaDoc for methods.

**Review by 1Mike3 — APPROVED** [review]
Requested changes Implemented,
Here's a Cookie for a Job Well done🍪

**Review by 1Mike3 — APPROVED** [review]
godd job,
here is your cookie🍪

**Review by 1Mike3 — APPROVED** [review]

**Review by 1Mike3 — APPROVED** [review]

**Review by 1Mike3 — APPROVED** [review]
approved

**Review by srebl — APPROVED** [review]

**Pull request merged**